sdk/core/Azure.Core/src/Serialization/ModelJsonConverter.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// .
+    /// </summary>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public class ModelJsonConverter : JsonConverter<IModelSerializable>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+    {
+        /// <summary>
+        /// .
+        /// </summary>
+        public bool IgnoreAdditionalProperties { get; set; } = true;
+
+        /// <summary>
+        /// .
+        /// </summary>
+        public ModelJsonConverter() { }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="ignoreAdditionalProperties"></param>
+        public ModelJsonConverter(bool ignoreAdditionalProperties)
+        {
+            IgnoreAdditionalProperties = ignoreAdditionalProperties;
+        }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="typeToConvert"></param>
+        /// <returns></returns>
+        public override bool CanConvert(Type typeToConvert)
+        {
+            return !Attribute.IsDefined(typeToConvert, typeof(JsonConverterAttribute));
+        }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="typeToConvert"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+        public override IModelSerializable Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+        {
+            return (IModelSerializable)ModelSerializer.Deserialize(BinaryData.FromString(JsonDocument.ParseValue(ref reader).RootElement.GetRawText()), typeToConvert, ConvertOptions(options));
```

Comment:
Could you use UTF-8 bytes here instead of a string?  Transcoding to UTF-16 adds a big cost overhead.